docs/examples/langchain/langchain_chat_pandas_df.py

```diff
@@ -0,0 +1,176 @@
+    """Will be asking the user for the API Key if not set as environment variable"""
+
+    OPENAI_API_KEY = param.String()
+
+
+class AgentConfig(param.Parameterized):
+    """Configuration used for the Pandas Agent"""
+
+    user = "Pandas Agent"
+    avatar = "🐼"
+
+    show_chain_of_thought = param.Boolean(default=False)
+
+    def _get_agent_message(self, message: str) -> pn.chat.ChatMessage:
+        return pn.chat.ChatMessage(message, user=self.user, avatar=self.avatar)
+
+
+class AppState(param.Parameterized):
+    data = param.DataFrame()
+
+    llm = param.Parameter(constant=True)
+    pandas_df_agent = param.Parameter(constant=True)
+
+    config: AgentConfig = param.ClassSelector(class_=AgentConfig)
+    environ: Environment = param.ClassSelector(class_=Environment, constant=True)
+
+    def __init__(
+        self, config: AgentConfig | None = None, environ: Environment | None = None
+    ):
+        if not config:
+            config = AgentConfig()
+        if not environ:
+            environ = Environment()
+        super().__init__(config=config, environ=environ)
+
+    @param.depends("environ.OPENAI_API_KEY", on_init=True, watch=True)
+    def _reset_llm(self):
+        with param.edit_constant(self):
+            if self.environ.OPENAI_API_KEY:
+                self.llm = ChatOpenAI(
+                    temperature=0,
+                    model="gpt-3.5-turbo-0613",
+                    api_key=self.environ.OPENAI_API_KEY,
+                    streaming=True,
+                )
+            else:
+                self.llm = None
+
+    @param.depends("llm", "data", on_init=True, watch=True)
+    def _reset_pandas_df_agent(self):
+        with param.edit_constant(self):
+            if not self.error_message:
+                self.pandas_df_agent = create_pandas_dataframe_agent(
+                    self.llm,
+                    self.data,
+                    verbose=True,
+                    agent_type=AgentType.OPENAI_FUNCTIONS,
+                    handle_parsing_errors=True,
+                )
+            else:
+                self.pandas_df_agent = None
+
+    @property
+    def error_message(self):
+        if not self.llm and self.data is None:
+            return """Please provide your `OPENAI_API_KEY`, **upload a `.csv` file** 
+and click the **send** button."""
+        if not self.llm:
+            return "Please provide your `OPENAI_API_KEY`."
+        if self.data is None:
+            return "Please **upload a `.csv` file** and click the **send** button."
+        return ""
+
+    @property
+    def welcome_message(self):
+        text = (
+            f"""I'm your <a href="\
+https://python.langchain.com/docs/integrations/toolkits/pandas" target="_blank">\
+LangChain Pandas DataFrame Agent</a>.
+
+I execute LLM generated Python code under the hood - this can be bad if the `llm`
+generated Python code is harmful. Use cautiously!
+
+{self.error_message}"""
+        ).strip()
```

Comment:
```suggestion
        text = dedent(f"""
            I'm your <a href="\
            https://python.langchain.com/docs/integrations/toolkits/pandas" target="_blank">\
            LangChain Pandas DataFrame Agent</a>.

            I execute LLM generated Python code under the hood - this can be bad if the `llm`
            generated Python code is harmful. Use cautiously!

            {self.error_message}"""
        ).strip()
```